### PR TITLE
fix: Instance comparison with `instanceof` is not realm-safe

### PR DESCRIPTION
**Pull request by mtrezza**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).

## Issue

Instance comparison with `instanceof` is not realm-safe. This may cause issues in certain test or production setups.

## Approach

- Replace occurrences of `instanceof` with safe alternatives.
- Add lint rule to prevent future use of `instanceof`.

## Tasks

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, code comments)
- [ ] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [ ] Add new Parse Error codes to Parse JS SDK 

## Summary by CodeRabbit

* **New Features**
  * Added cross-realm-safe type-checking utilities for Date, RegExp, Map, Set, native Error, Promise/thenable, and generic object detection.

* **Improvements**
  * Replaced fragile instanceof checks with the new, safer utilities and Array.isArray/Buffer.isBuffer where appropriate for more reliable runtime behavior.

* **Tests**
  * Expanded unit tests to cover cross-realm, edge-case, and prototype-pollution scenarios.

* **Chores**
  * Added lint rules to enforce use of the safer type-check utilities.

### CODE REVIEW

**Comment by parseplatformorg** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/acinader/pr-checks/0846b635-1ee1-4b63-a8d4-d75fcc173e23?refs=837692d28581e4c3837c498cd7a618e3cc25c5ba&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: spec/Utils.spec.js [hunk 1]
```diff
@@ -287,4 +288,159 @@ describe('Utils', () => {
...
+        expect(Utils.isPromise({})).toBe(false);
+        expect(Utils.isPromise({ a: 1 })).toBe(false);
+      } finally {
+        delete Object.prototype.then;
+      }
+    });
+    it('should return true for real thenables even when Object.prototype.then is polluted', () => {
+      Object.prototype.then = () => {};
+      try {
+        expect(Utils.isPromise({ then: () => {} })).toBe(true);
```
[the remark this answers is not in the document]
Comment (mtrezza) [reply spec/Utils.spec.js:419]: The project doesn't use Biome at all.

File: spec/Utils.spec.js [hunk 2]
```diff
@@ -287,4 +288,161 @@ describe('Utils', () => {
...
+
+  describe('isSet', () => {
+    it('should return true for a Set', () => {
+      expect(Utils.isSet(new Set())).toBe(true);
+    });
+    it('should return true for a cross-realm Set', () => {
+      const crossRealmSet = vm.runInNewContext('new Set()');
+      expect(crossRealmSet instanceof Set).toBe(false);
+      expect(Utils.isSet(crossRealmSet)).toBe(true);
+    });
```
[the remark this answers is not in the document]
Comment (mtrezza) [reply spec/Utils.spec.js:357]: Check to see whether the issue is addressed.

**Comment by mtrezza** [comment]
@coderabbitai resume

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [9.6.0-alpha.32](https://github.com/parse-community/parse-server/releases/tag/9.6.0-alpha.32)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [9.6.0](https://github.com/parse-community/parse-server/releases/tag/9.6.0)
